Switch more tests to zen2

andrershov opened the pull request:
1. CCR tests work without any changes
2. `testDanglingIndices` require changes the source code (added TODO).
3. `testIndexDeletionWhenNodeRejoins` because it's using just two
nodes, adding the node to exclusions is needed on restart.
4. `testCorruptTranslogTruncationOfReplica` starts dedicated master
one, because otherwise, the cluster does not form, if nodes are stopped
and one node is started back.
5. `testResolvePath` needs TEST cluster, because all nodes are stopped
at the end of the test and it's not possible to perform checks needed
by SUITE cluster.
6. `SnapshotDisruptionIT`. Without changes, the test fails because Zen2
retries snapshot creation as soon as network partition heals. This
results into the race between creating snapshot and test cleanup logic
(deleting index). Zen1 on the
other hand, also schedules retry, but it takes some time after network
partition heals, so cleanup logic executes latter and test passes. The
check that snapshot is eventually created is added to
the end of the test.

ywelsch commented on server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java:
```diff
@@ -328,14 +335,14 @@ public boolean clearData(String nodeName) {
     public void testIndexDeletionWhenNodeRejoins() throws Exception {
         final String indexName = "test-index-del-on-node-rejoin-idx";
-        final int numNodes = 2;
+        // We need at least 3 nodes to make sure, that once one node is stopped, remaining nodes can elect a new master
```
I think the proper fix here is to add the down-scale / up-scale logic that's in InternalTestCluster#stopNodesAndClients to restartNode as well, just as we do with the autoManageMinMasterNodes scaling. The test should then not need any adaptation

andrershov replied:
https://github.com/elastic/elasticsearch/pull/36367/commits/6e5bbd4cc235aeadbf7a72d2f4642635c98690ec

ywelsch commented on server/src/test/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommandIT.java:
```diff
@@ -555,8 +546,10 @@ public Settings onNodeStopped(String nodeName) throws Exception {
     }
 
     public void testResolvePath() throws Exception {
+        internalCluster().startMasterOnlyNode();
```
I think this one can be fixed by moving the scope to ESIntegTestCase.Scope.TEST. As these tests are starting up nodes anyway, there's no need to keep the cluster per suite. Maybe that also fixes the testCorruptTranslogTruncationOfReplica

andrershov replied:
https://github.com/elastic/elasticsearch/pull/36367/commits/4d9d141b90c390068f1374a471309e7e8d7cf006 this helps this test but does not help another test in this class (it has different root cause)

andrershov commented on server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java:
```diff
@@ -276,11 +276,18 @@ public void testTwoNodesSingleDoc() throws Exception {
     }
 
     public void testDanglingIndices() throws Exception {
+        /*TODO This test test does not work with Zen2, because once master node looses its cluster state during restart
```
https://github.com/elastic/elasticsearch/pull/36367/commits/830def10fdc55b5e40b4b8ebe315c150491786de

andrershov commented:
@ywelsch I've addressed your comments + I've also fixed `SnapshotDisruptionIT` (see PR description) with https://github.com/elastic/elasticsearch/pull/36367/commits/edfa48ad9ce79b7e3b3bf0d10789d88bc359e227. Can you please take a look?

ywelsch commented on test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java:
```diff
@@ -1746,31 +1711,78 @@ public synchronized void rollingRestart(RestartCallback callback) throws Excepti
         }
-        if (callback.validateClusterForming() || updateMinMaster) {
+
+        if (callback.validateClusterForming() && excludedNodeIds.isEmpty() == false) {
```
This should be an `||`, not an `&&`?
